Move Grafana setting files to Persistent Storage (#84)
antrea-io/theia: docs/network-flow-visibility.md

### Dashboard Customization

If you would like to make any change to any of the pre-built dashboards, or build
a new dashboard, please follow this [doc](https://grafana.com/docs/grafana/latest/dashboards/)
on how to build a dashboard.

From Theia 0.2, we use PersistentVolume for the Grafana configuration database.
If you create a new dashboard or modify the pre-built dashboards, once you click
on the "Save dashboard" button, the changes will be kept after a Grafana
Deployment restart. Other changes in settings like passwords and preferences will
also be kept.

> In Theia 0.1, the dashboards will be lost after restarting the Grafana Deployment.

Reviewer: ```suggestion
In Theia 0.1, the changes to dashboards and settings will be lost after restarting the Grafana Deployment.
```

Author: Updated. Thanks!

> To restore those changes after a restart, as the first step, you will need to

Reviewer: ```suggestion
To restore those changes to dashboards after a restart, as the first step, you will need to
```

### Dashboard Customization

If you would like to make any change to any of the pre-built dashboards, or build
a new dashboard, please follow this [doc](https://grafana.com/docs/grafana/latest/dashboards/)
on how to build a dashboard.

From Theia 0.2, we use PersistentVolume for the Grafana configuration database.
If you create a new dashboard or modify the pre-built dashboards, once you click
on the "Save dashboard" button, the changes will be kept after a Grafana

> Deployment restart. Other changes in settings like passwords and preferences will

Reviewer 2: Since grafana-pv is a hostPath volume I think we should clarify that setting are stored as long as the grafana pod is deployed on the same host. In order to make sure settings are preserved regardless of where the grafana pod is deployed, users should adjust grafana-pv configuration to use nfs or some other form of shared storage.

Author: Got it. Do you think we could use the similar PV solution as for ClickHouse, which allows user to choose a type of the PersistentVolume created by us or provide `persistentVolumeClaimSpec`? We can set `HostPath` as the default type and support Local and NFS options.

Author: I add a patch to support customized Grafana PV and update the doc. Please take a look if it is better.

Reviewer 2: Great job!

on how to build a dashboard.

By clicking on the "Save dashboard" button in the Grafana UI, the changes to the

> dashboards will be saved in the Grafana database.

Reviewer: This paragraph can be removed, with its content combined with the next paragraph.

> From Theia 0.2, Grafana uses PersistentVolume for the Grafana database. If you

Reviewer: ```suggestion
From Theia 0.2, we use PersistentVolume for the Grafana configuration database. If you
```

> create a new dashboard or modify the pre-built dashboards, the changes will be

Reviewer: ```suggestion
create a new dashboard or modify the pre-built dashboards, once you click on the "Save dashboard" button, the changes will be
```

> kept after a restart. Other changes in settings like passwords and preferences

Author: Thanks Anlan for reviewing. Updated!

> can also be persistent.

Reviewer: ```suggestion
will also be kept.
```

> In Theia 0.1, the dashboards will be lost after restarting the Grafana deployment.

Reviewer: ```suggestion
In Theia 0.1, the dashboards will be lost after restarting the Grafana Deployment.
```
